src/durableorchestrationclient.ts

```diff
@@ -362,39 +341,55 @@ export class DurableOrchestrationClient {
         eventData: unknown,
         taskHubName?: string,
         connectionName?: string,
-        ): Promise<void> {
+    ): Promise<void> {
         if (!eventName) {
             throw new Error("eventName must be a valid string.");
         }
 
-        const idPlaceholder = this.clientData.managementUrls.id;
-        let requestUrl = this.clientData.managementUrls.sendEventPostUri
-            .replace(idPlaceholder, instanceId)
-            .replace(this.eventNamePlaceholder, eventName);
+        let requestUrl: string;
+        if (this.clientData.rpcBaseUrl) {
+            // Fast local RPC path
+            let path = `instances/${instanceId}/raiseEvent/${eventName}`;
```

Comment:
I'm intentionally hardcoding the URL paths here rather than doing templating or variable replacement. I just didn't feel the potential benefit was worth the added complexity. Basically, I'm considering the HTTP API to be a well-known contract and I would expect other language implementations to follow this model as well.

Reply:
I'm 100% OK with this. I tried to move closer to this direction with the new Entity APIs, because the maintenance cost of adding a new template everytime we wanted to support a new API seemed not worth the cost.

In fact, in order to keep our legacy/local HTTP logic as close as possible, I would recommend extracting the base url and authorization key from one of the existing templates. Then the path building and query string building logic in the two code paths can be almost entirely shared, and we don't have to maintain any legacy templating logic.

Reply:
Do you think it's really necessary for us to make any changes to the legacy code? Per my other comment, I didn't see any need to maintain it at all, which is why I opted to leave it untouched.